### PR TITLE
RUN-2994 close hanging window

### PR DESCRIPTION
**Pull request by datamadic**
Do not subscribe directly to the unload event.
[tests well](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5951638f0c101702173658a8) fails pass in isolation
[added test](https://testing-dashboard.openfin.co/#/app/tests/595113230c10170217365894/edit)

### CODE REVIEW

**Comment by HarsimranSingh** [comment]
👍

File: src/renderer/api-decorator.js [hunk 1]
```diff
@@ -342,6 +342,16 @@ limitations under the License.
         }, 1);
     }
 
+    rqr('electron').remote.getCurrentWebContents(renderFrameId).once('navigation-entry-commited', () => {
```
Comment (joneit) [inline src/renderer/api-decorator.js]: With this addition, `rqr('electron')` appears 5 times in this file now, lets add an `electron` variable at top of file:
```js
    let rqr = require;
    let electron = rqr('electron');
    let webFrame = electron.webFrame.createForRenderFrame(renderFrameId);
    let ipc = electron.ipcRenderer;
    // etc.!
```

**Comment by datamadic** [comment]
@joneit  have a look, removed redundant requires. 
 
[updated tests](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/595571f70c1017021736591e)

File: src/renderer/api-decorator.js [hunk 2]
```diff
@@ -28,9 +28,9 @@ limitations under the License.
     let isMainRenderFrame = global.isMainFrame;
     let glbl = global;
 
-    let rqr = require;
-    let webFrame = rqr('electron').webFrame.createForRenderFrame(renderFrameId);
-    let ipc = rqr('electron').ipcRenderer;
+    let electron = require('electron');
+    let webFrame = electron.webFrame.createForRenderFrame(renderFrameId);
+    let ipc = electron.ipcRenderer;
```
Comment (whyn07m3) [inline src/renderer/api-decorator.js]: Since lines 31,32 and 33 are changing, `const`s? :)

Comment (datamadic) [inline src/renderer/api-decorator.js]: you want `const`? you got it. give me a min

**Pull request merged**